bwdata.py

```diff
@@ -0,0 +1,805 @@
+"""
+bwrdata contains the BWData class.
+"""
+import datetime
+import filters
+
+class BWData:
+    """
+    This class is a superclass for brandwatch BWQueries and BWGroups.  It was built to handle resources that access data (e.g. mentions, topics, charts, etc).
+    """
+    def get_mentions(self, name=None, startDate=None, max_pages=None, **kwargs):
+        """
+        Retrieves a list of mentions.
+        Note: Clients do not have access to full Twitter mentions through the API because of our data agreement with Twitter.
+
+        Args:
+            name:       You must pass in a query / group name (string).
+            startDate:  You must pass in a start date (string).
+            max_pages:  Maximum number of pages to retrieve, where each page is 5000 mentions by default - Optional.  If you don't pass max_pages, it will retrieve all mentions that match your request.
+            kwargs:     All other filters are optional and can be found in filters.py.
+
+        Raises:
+            KeyError:   If the mentions call fails.
+
+        Returns:
+            A list of mentions.
+        """
+        params = self._fill_params(name, startDate, kwargs)
+        params["pageSize"] = kwargs["pageSize"] if "pageSize" in kwargs else 5000
+        params["page"] = kwargs["page"] if "page" in kwargs else 0
+        all_mentions = []
+
+        while max_pages == None or params["page"] < max_pages:
+            next_mentions = self._get_mentions_page(params, params["page"])
+
+            if len(next_mentions) > 0:
+                all_mentions += next_mentions
+
+                if self.console_report:
+                    print("Page " + str(params["page"]) + " of " + self.resource_type + " " + name + " retrieved")
+            else:
+                break
+
+            params["page"] += 1
+
+        if self.console_report:
+            print(str(len(all_mentions)) + " mentions downloaded")
+        return all_mentions
+
+    def num_mentions(self, name=None, startDate=None, **kwargs):
+        """
+        Retrieves a count of the mentions in a given timeframe.
+
+        Args:
+            name:       You must pass in a query / group name (string).
+            startDate:  You must pass in a start date (string).
+            kwargs:     All other filters are optional and can be found in filters.py.
+
+        Returns:
+            A count of the mentions in a given timeframe.
+        """
+        params = self._fill_params(name, startDate, kwargs)
+        return self.project.get(endpoint="data/mentions/count", params=params)
```

Comment:
The `/data/mentions/count` call returns a JSON of the form `{"mentionsCount": <the actual stuff the user wants>}`.  Could we modify this function to return `self.project.get(...)["mentionsCount"]`?  (This PR seems as good a time as any to fix this.)